src/ophyd_async/core/_mock_signal_utils.py

```diff
@@ -8,11 +8,12 @@
 
 
 def _get_mock_signal_backend(signal: Signal) -> MockSignalBackend:
-    assert isinstance(signal._backend, MockSignalBackend), (
+    backend = signal.get_backend()
```

Comment:
I've got 2 suggestions here, either:
- Add a `#noqa` and leave the private access
- When a mock signal backend is created, put it in a global dict, and grab it out of that dict in this function

Reply:
I'll just do a `#noqa`, its only one place.